src/providers/pdal/qgspdalprovider.cpp

```diff
@@ -58,8 +61,82 @@ QgsRectangle QgsPdalProvider::extent() const
 
 QgsPointCloudAttributeCollection QgsPdalProvider::attributes() const
 {
-  // TODO
-  return QgsPointCloudAttributeCollection();
+  return mIndex->attributes();
+}
+
+QVariantList QgsPdalProvider::metadataClasses( const QString &attribute ) const
+{
+  return mIndex->metadataClasses( attribute );
+}
+
+QVariant QgsPdalProvider::metadataClassStatistic( const QString &attribute, const QVariant &value, QgsStatisticalSummary::Statistic statistic ) const
+{
+  return mIndex->metadataClassStatistic( attribute, value, statistic );
+}
+
+
+
+void QgsPdalProvider::loadIndex( bool skipIndexGeneration )
+{
+  if ( !mIsValid )
```

Comment:
Is this condition correct? Because of this condition, indexing was not starting for me at all... When I changed it, then it started to run untwine...
```suggestion
  if ( mIsValid )
```

Reply:
Hmm and now when I have `if ( mIsValid )` it re-indexes my files every time :-)